softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -718,6 +765,29 @@ func resourceSoftLayerVirtualGuestRead(d *schema.ResourceData, meta interface{})
 	}
 	d.SetConnInfo(connInfo)
 
+	// Read secondary IP addresses
+	d.Set("secondary_ip_addresses", nil)
+	if result.PrimaryIpAddress != nil {
+		secondarySubnetResult, err := services.GetAccountService(meta.(ProviderConfig).SoftLayerSession()).
+			Mask("ipAddresses[id,ipAddress]").
+			Filter(filter.Build(filter.Path("publicSubnets.endPointIpAddress.ipAddress").Eq(*result.PrimaryIpAddress))).
+			GetPublicSubnets()
+		if err != nil {
+			log.Printf("Error getting secondary Ip addresses: %s", err)
+		}
+
+		secondaryIps := make([]string, 0)
```

Comment:
if we know the number of secondary ips ahead of time, we can set the capacity in this make statement. `make([]string, 0, len(secondarySubnetResult))`. It is a small optimization.